.mage/js.go

```diff
@@ -60,17 +71,265 @@ func (js Js) yarn() (func(args ...string) error, error) {
+	out := os.Stdout
+	if !mg.Verbose() {
+		out = nil
+	}
+	return func(args ...string) error {
+		_, err := sh.Exec(nil, out, os.Stderr, nodeBin(cmd), args...)
+		return err
+	}, nil
+}
+
+func (js Js) webpack() (func(args ...string) error, error) {
+	return js.execFromNodeBin("webpack")
+}
+
+func (js Js) webpackServe() (func(args ...string) error, error) {
+	return js.execFromNodeBin("webpack-dev-server")
+}
+
+func (js Js) babel() (func(args ...string) error, error) {
+	if _, err := os.Stat(nodeBin("babel")); os.IsNotExist(err) {
+		if err = js.DevDeps(); err != nil {
+			return nil, err
+		}
+	}
+	return func(args ...string) error {
+		_, err := sh.Exec(nil, nil, os.Stderr, nodeBin("babel"), args...)
+		return err
+	}, nil
+}
+
+func (js Js) jest() (func(args ...string) error, error) {
+	return js.execFromNodeBin("jest")
+}
+
+func (js Js) eslint() (func(args ...string) error, error) {
+	return js.execFromNodeBin("eslint")
+}
+
+func (js Js) stylint() (func(args ...string) error, error) {
+	return js.execFromNodeBin("stylint")
+}
+
+func (js Js) storybook() (func(args ...string) error, error) {
+	return js.execFromNodeBin("start-storybook")
+}
+
 // DevDeps installs the javascript development dependencies.
 func (js Js) DevDeps() error {
-	_, err := js.yarn()
-	return err
+	return devDeps()
 }
 
 // Deps installs the javascript dependencies.
 func (js Js) Deps() error {
-	yarn, err := js.yarn()
+	if mg.Verbose() {
+		fmt.Println("Installing JS dependencies")
+	}
+	yarn, err := yarn()
+	if err != nil {
+		return err
+	}
+	err = yarn("install", "--no-progress", "--production=false")
+	if err != nil {
+		return err
+	}
+	mg.Deps(JsSDK.Link)
+	return nil
+}
+
+// Build runs all necessary commands to build the console bundles and files.
+func (js Js) Build() {
+	mg.SerialDeps(js.Deps, JsSDK.Build, js.BuildDll, js.BuildMain)
+}
+
+// BuildMain runs the webpack command with the project config.
+func (js Js) BuildMain() error {
+	mg.Deps(js.Translations, js.BackendTranslations, js.BuildDll)
+	if mg.Verbose() {
+		fmt.Println("Running Webpack")
+	}
+	webpack, err := js.webpack()
+	if err != nil {
+		return err
+	}
+	return webpack("--config", "config/webpack.config.babel.js")
+}
+
+// BuildDll runs the webpack to build the DLL bundle
+func (js Js) BuildDll() error {
+	changed, err := target.Path("./public/libs.bundle.js", "./yarn.lock")
+	if os.IsNotExist(err) || (err == nil && changed) {
+		if mg.Verbose() {
+			fmt.Println("Running Webpack for DLL…")
+		}
+		webpack, err := js.webpack()
+		if err != nil {
+			return err
+		}
+		return webpack("--config", "config/webpack.dll.babel.js")
+	}
+	return nil
+}
+
+// Serve builds necessary bundles and serves the console for development.
+func (js Js) Serve() {
+	mg.Deps(js.ServeMain)
+}
+
+// ServeMain runs webpack-dev-server
+func (js Js) ServeMain() error {
+	mg.Deps(js.Translations, js.BackendTranslations, js.BuildDll)
+	if mg.Verbose() {
+		fmt.Println("Running Webpack for Main Bundle in watch mode…")
```

Comment:
I think we should always display this message. At the moment, after running `mage js:serve` there is no output at all while the terminal is blocked by the `webpack-dev-server` process. the only reason to use this is for development purposes and should not bother developers who just want to deploy the stack and keep output clean.

Reply:
I agree. Had a discussion about this as well with @htdvisser, who said (correct me if I'm wrong) that we should for now rather follow the convention and show target output only when mage's verbose mode `-v` is enabled. Overall the output of targets is not really consistent yet. We should look into this once more.

Reply:
I have piped the output for these serve targets back to `os.Stdout` by default. @htdvisser feel free to drop `51dc0bf` if you disagree.